pm/sendermonitor.go

```diff
@@ -157,82 +158,75 @@ func (sm *senderMonitor) SubFloat(addr ethcommon.Address, amount *big.Int) error
 	// An updated max float results in updated ticket params
 	// The sender could plausibly send tickets that trigger acceptable errors
 	sm.em.ClearErrCount(addr)
-
-	return nil
 }
 
 // MaxFloat returns a remote sender's max float
 func (sm *senderMonitor) MaxFloat(addr ethcommon.Address) (*big.Int, error) {
 	sm.mu.RLock()
 	defer sm.mu.RUnlock()
 
-	if err := sm.ensureCache(addr); err != nil {
-		return nil, err
-	}
+	sm.ensureCache(addr)
 
-	return sm.maxFloat(addr), nil
+	return sm.maxFloat(addr)
 }
 
 // QueueTicket adds a ticket to the queue for a remote sender
-func (sm *senderMonitor) QueueTicket(addr ethcommon.Address, ticket *SignedTicket) error {
+func (sm *senderMonitor) QueueTicket(addr ethcommon.Address, ticket *SignedTicket) {
 	sm.mu.Lock()
 	defer sm.mu.Unlock()
 
-	if err := sm.ensureCache(addr); err != nil {
-		return err
-	}
+	sm.ensureCache(addr)
 
 	sm.senders[addr].queue.Add(ticket)
-
-	return nil
 }
 
 // maxFloat is a helper that returns the sender's max float as:
 // reserveAlloc - pendingAmount
 // Caller should hold the lock for senderMonitor
-func (sm *senderMonitor) maxFloat(addr ethcommon.Address) *big.Int {
-	return new(big.Int).Sub(sm.senders[addr].reserveAlloc, sm.senders[addr].pendingAmount)
+func (sm *senderMonitor) maxFloat(addr ethcommon.Address) (*big.Int, error) {
+	reserveAlloc, err := sm.reserveAlloc(addr)
+	if err != nil {
+		return nil, err
+	}
+	return reserveAlloc.Sub(reserveAlloc, sm.senders[addr].pendingAmount), nil
```

Comment:
Might be worth allocating a new big.Int here instead of mutating `reserveAlloc` to make sure we never accidentally mutate a value we're not supposed to (not a problem now since `reserveAlloc()` returns a fresh big.Int